afs/afs-ws/afs-ws-storage/src/main/java/com/powsybl/afs/ws/storage/RemoteAppStorage.java

```diff
@@ -0,0 +1,892 @@
+/**
+ * Copyright (c) 2018, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.afs.ws.storage;
+
+import com.google.common.io.ByteStreams;
+import com.google.common.util.concurrent.UncheckedExecutionException;
+import com.powsybl.afs.storage.AppStorage;
+import com.powsybl.afs.storage.NodeDependency;
+import com.powsybl.afs.storage.NodeGenericMetadata;
+import com.powsybl.afs.storage.NodeInfo;
+import com.powsybl.afs.storage.buffer.StorageChangeBuffer;
+import com.powsybl.commons.exceptions.UncheckedInterruptedException;
+import com.powsybl.commons.io.ForwardingInputStream;
+import com.powsybl.commons.io.ForwardingOutputStream;
+import com.powsybl.math.timeseries.*;
+import com.powsybl.afs.ws.client.utils.ClientUtils;
+import com.powsybl.afs.ws.utils.AfsRestApi;
+import com.powsybl.afs.ws.utils.JsonProvider;
+import com.powsybl.afs.ws.utils.gzip.ReaderInterceptorGzip;
+import com.powsybl.afs.ws.utils.gzip.WriterInterceptorGzipCli;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.client.AsyncInvoker;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.*;
+import java.io.*;
+import java.net.URI;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+import static com.powsybl.afs.ws.client.utils.ClientUtils.*;
+
+/**
+ * @author Ali Tahanout <ali.tahanout at rte-france.com>
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+public class RemoteAppStorage implements AppStorage {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RemoteAppStorage.class);
+
+    private static final int BUFFER_MAXIMUM_CHANGE = 1000;
+    private static final long BUFFER_MAXIMUM_SIZE = Math.round(Math.pow(2, 20)); // 1Mo
```

Comment:
Seriously? Do we really need to compute the buffer size like this?

Reply:
Yes